### PR TITLE
refactor: create test_base_cluster

### PR DESCRIPTION
**Pull request by rafilong**
Towards #78
Stacked on #79

### CODE REVIEW

File: tests/unit/test_cluster.py [hunk 1]
```diff
@@ -19,48 +19,15 @@
 
 from ._testing import _make_credentials
 
+from tests.unit.test_base_cluster import (
```
Comment (kolea2) [inline tests/unit/test_cluster.py:22]: what's this change for?

Comment (rafilong) [reply tests/unit/test_cluster.py:22]: An attempt at deduping the various cluster test classes. `_Instance` and `_Client` implementation are shared between testers, and constants are the same regardless of sync/async.

Comment (rafilong) [reply tests/unit/test_cluster.py:22]: From what I saw it seems pretty feasible to declare a global helper class for all testers and import `_Instance` and `_Client` from there, but this seemed like a good middle-ground towards that. Happy to make that change in a separate PR though, or to keep it as-is if there's a reason for the current implementation.

Comment (kolea2) [reply tests/unit/test_cluster.py:22]: nope sounds good to me! Thanks for the explanation

**Pull request merged**